addons/mail/models/mail_thread.py

```diff
@@ -1152,6 +1152,8 @@ def _message_route_process(self, message, message_dict, routes):
                 thread_id = thread.id
                 subtype_id = thread._creation_subtype().id
 
+            # switch to odoobot for all incoming message creation
+            thread = thread.with_user(self.env.ref('base.user_root'))
```

Comment:
Yes, that's actually why I use odoobot instead of the user.

If we don't do this, calling message_process with some other user non-archived user will lead to the same issue. This is notably the case when:

- Clicking 'fetch now' in the incoming server settings
- Using a script that connects as some active user to call message_process to process .eml files directly (Our mail-gateway.py does this)

With this, we are assured the messages are posted with an inactive user and that our assumptions for authorship hold.

Reply:
Isn't it the role of the caller to maybe use sudo() before calling the message_process methods ?

Reply:
With the current imp for 'real_author', which checks whether the user itself is active, this won't work because even if you sudo the current user doesn't actually change. You just set the su variable to bypass security checks.

'thread' is already sudoed before hand thanks to `ModelCtx = Model.with_user(related_user).sudo()` so switching to root entirely should be fine security-wise, or at least it will be as good as it currently is.

Reply:
Note that we could also / or detect a su environment in the "real author" detection stuff :D . Would prefer seeing if we can do something with that, instead of the hard usage of "root" here that I don't really like.

Reply:
Mm, I'm just worried that instances of 'sending a message as someone else' which the problematic feature addressed would also often involved sudo-ing which then goes against the point if we ignore the feature when using sudo

We can use context keys if you prefer but uh.. it would require adding a context key

Reply:
All in all, ok with the usage of root then (just rename variable as discussed).